src/libmongoc/src/mongoc/mongoc-bulk-operation-private.h

```diff
@@ -33,6 +33,7 @@ struct _mongoc_bulk_operation_t {
    mongoc_client_session_t *session;
    mongoc_write_concern_t *write_concern;
    mongoc_bulk_write_flags_t flags;
+   bson_t *let;
```

Comment:
```suggestion
   bson_t let;
```
I am not seeing a compelling reason why the `bson_t` object should be dynamically allocated instead of being just a plain data member. `bulk->let = bson_copy (let);` in `mongoc_bulk_operation_set_let` could be made `bson_copy_to (let, &bulk->let);`, and `!bson_empty0 (bulk->let)` checks could be made `!bson_empty (&bulk->let)`.

Reply:
Changed to an embedded `bson_t` data member. I also ensure that we initialize the field with `BSON_INITIALIZER` in `mongoc_bulk_operation_new`.